java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -7384,6 +7385,129 @@ public Long scheduleApplyHighstate(User loggedInUser, Integer sid, Date earliest
             throw new TaskomaticApiException(e.getMessage());
         }
     }
+
+    /**
+     * Schedule highstate application for a list of systems.
+     *
+     * @param loggedInUser The current user
+     * @param sids The list of system id of the target systems
+     * @param earliestOccurrence Earliest occurrence
+     * @param test Run states in test-only mode
+     * @return list of action id or exception thrown otherwise
+     *
+     * @xmlrpc.doc Schedule highstate application for a given system.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #array_single("long", "systemIds")
+     * @xmlrpc.param #param("dateTime.iso8601", "earliestOccurrence")
+     * @xmlrpc.param #param_desc("boolean", "test", "Run states in test-only mode")
+     * @xmlrpc.returntype #param("int", "actionId", "The action id of the scheduled action")
+     */
+    public Long scheduleApplyHighstate(User loggedInUser, List<Long> sids, Date earliestOccurrence, boolean test) {
```

Comment:
Just like @parlt91 commented below, maybe the previous method (`scheduleApplyHighstate(User loggedInUser, Integer sid, Date earliestOccurrence, boolean test)`) could just call this one with a list containing just one `sid`?

Reply:
done